generator.py

```diff
@@ -78,102 +83,72 @@ def regen_crates(path, conf, level=0):
+    parser.add_argument('--gir-files', default=DEFAULT_GIR_FILES,
+                        help='Path of the gir-files folder')
+    parser.add_argument('--gir-path', default=DEFAULT_GIR_PATH,
+                        help='Path of the gir executable to run')
+    parser.add_argument('--yes', action='store_true',
+                        help=' Always answer `yes` to any question asked by the script')
+    parser.add_argument('--no-fmt', action='store_true',
+                        help='If set, this script will not run `cargo fmt`')
 
-def parse_args(args):
-    conf = {
-        "gir_path": None,
-        "gir_files": None,
-        "yes": False,
-        "run_fmt": True,
-    }
-    i = 0
-
-    while i < len(args):
-        arg = args[i]
-        if arg == "-h" or arg == "--help":
-            print_help()
-            return None
-        elif arg == "--gir-path":
-            i += 1
-            if i >= len(args):
-                print("Expected argument after `--gir-path` option...")
-                return None
-            if not isfile(args[i]):
-                print("`{}` file doesn't exist. Aborting...".format(args[i]))
-                return None
-            conf["gir_path"] = args[i]
-        elif arg == "--gir-files":
-            i += 1
-            if i >= len(args):
-                print("Expected argument after `--gir-files` option...")
-                return None
-            if not isdir(args[i]):
-                print("`{}` folder doesn't exist. Aborting...".format(args[i]))
-                return None
-            conf["gir_files"] = args[i]
-        elif arg == "--yes":
-            conf["yes"] = True
-        elif arg == "--no-fmt":
-            conf["run_fmt"] = False
-        else:
-            print("Unknown argument `{}`.".format(arg))
-            return None
-        i += 1
-    return conf
+    return parser.parse_args()
 
 
 def main():
     gir_path = None
 
     conf = parse_args(sys.argv[1:])
-    if conf is None:
-        return 1
+    print(conf)
 
-    if conf["gir_files"] is None:
+    if conf.gir_files == DEFAULT_GIR_FILES:
         if def_check_submodule("gir-files", conf) == FAILURE:
             return 1
-        conf["gir_files"] = "gir-files"
-    if conf["gir_path"] is None:
+    else:
+       if not isdir(conf.gir_files):
+            print("`{}` dir doesn't exist. Aborting...".format(path))
+            return 1
+
+    if conf.gir_path == DEFAULT_GIR_PATH:
         if not build_gir_if_needed(def_check_submodule("gir", conf)):
             return 1
-        conf["gir_path"] = "./gir/target/release/gir"
+    else:
+       if not isfile(conf.gir_path):
+            print("`{}` file doesn't exist. Aborting...".format(path))
+            return 1
 
     print('=> Regenerating crates...')
-    if not regen_crates(".", conf):
-        return 1
-    if conf["run_fmt"] is True and not run_command(['cargo', 'fmt']):
+    for path in conf.path:
```

Comment:
Nice add!

Reply:
That's the only thing I wanted to change and I thought it would be simpler using argparse xD